src/core/connection.c

```diff
@@ -1322,6 +1322,9 @@ QuicConnOnShutdownComplete(
     Connection->State.ShutdownComplete = TRUE;
     Connection->State.UpdateWorker = FALSE;
 
+    // call earlier then others for frames which requires ClientCallbackHandler to freeing app memory (Datagram)
+    QuicLossDetectionUninitialize(&Connection->LossDetection);
```

Comment:
Instead of moving this up, let's try moving the shutdown complete event down, after send uninitialize?